libs/sdk-ui-filters/src/AttributeFilter/hooks/useAttributeFilterController.ts

```diff
@@ -311,6 +311,9 @@ function updateNonResettingFilter(
         setConnectedPlaceholderValue(nextFilter);
 
         if (supportsKeepingDependentFiltersSelection && hasNumberOfLimitingAttributesChanged) {
+            handler.changeSelection({ ...handler.getWorkingSelection(), irrelevantKeys: [] });
```

Comment:
Just curious, is it fine to call handler.changeSelection multiple times?  What about removing from here and just "appending" to the previous changeSelection command?

what about something like (pseudocode, not tested)
```
        const shouldEmptyIrrelevantKeys = supportsKeepingDependentFiltersSelection && hasNumberOfLimitingAttributesChanged; // TODO: find better variable name
        const addToSelection = shouldEmptyIrrelevantKeys ? { irrelevantKeys: [] } : {}
        handler.changeSelection({ keys, isInverted, ...addToSelection });
        handler.setLimitingAttributeFilters(limitingAttributeFilters);
...
        if (shouldEmptyIrrelevantKeys) {
            setShouldReloadElements(true);
            return "init-self";
        }
```

Reply:
Updated, thanks